pyxem/utils/indexation_utils.py

```diff
@@ -38,18 +38,96 @@
 OrientationResult = namedtuple("OrientationResult",
                                "phase_index rotation_matrix match_rate error_hkls total_error scale center_x center_y".split())
 
+#Functions used in correlate_library.
+def fast_correlation(image_intensities,int_local,pn_local, **kwargs):
+    """
+    Computes the correlation score between an image and a template, using the formula
+    .. math:: FastCorrelation
+        \\frac{\\sum_{j=1}^m P(x_j, y_j) T(x_j, y_j)}{\\sqrt{\\sum_{j=1}^m T^2(x_j, y_j)}}
+
+    Parameters
+    ----------
+    image_intensities: list
+        list of intensity values in the image, for pixels where the template has a non-zero intensity
+     int_local: list
+        list of all non-zero intensities in the template
+     pn_local: float
+        pattern norm of the template
+
+    Returns
+    -------
+    corr_local: float
+        correlation score between template and image.
+
+    See also:
+    ---------
+    correlate_library, normalized_correlation
+
+    """
+    return np.sum(np.multiply(image_intensities, int_local)) / \
+                pn_local  # Correlation is the partially normalized dot product
+
+def normalized_correlation(nb_pixels,image_std,average_image_intensity,image_intensities,int_local, **kwargs):
```

Comment:
I am fairly convinced that this should be called "zero_mean_normalized_correlation" & updates elsewhere to reflect this.